armory/scenarios/scenario.py

```diff
@@ -313,16 +319,24 @@ def run_attack(self):
             )
         self.metrics_logger.update_perturbation(x, x_adv)
 
-        if self.sample_exporter is not None:
-            self.sample_exporter.export(x, x_adv, y, y_pred_adv)
-
         self.x_adv, self.y_target, self.y_pred_adv = x_adv, y_target, y_pred_adv
 
+    def export_samples(self):
+        self.sample_exporter.export(
```

Comment:
If x and x_adv is None, how does this behave? Should we raise a user-facing Error to the effect of "Must call `evaluate_current` before exporting samples?

Reply:
it will properly handle if `x_adv` is `None` (e.g. when `--skip-attack` is set). `x` will never be `None` when running a scenario, unless I suppose a user interactively ran the following:

```
from armory import scenarios
s = scenarios.get("/armory/tmp/2022-03-15T202227.880004/interactive-config.json").load()
s.export_samples()
```

Is this the scenario where you envisioned `x` could be `None`? Here's the error that arises in this case:

```
AttributeError: 'ImageClassificationTask' object has no attribute 'x'
```

That's somewhat cryptic of an error message, although I think this is an unlikely edge case to occur in the first place. That said, I added a check in 6339176 which instead shows:

```
AttributeError: ImageClassificationTask has no attribute 'x'. Be sure to call evaluate_current() before attempting to export samples. 

```

Reply:
No, I was considering a scenario where we have an adversarial dataset with `x_adv` but no `x`. Both being None would be weird.

As an aside, I like the check.